core/src/price_estimation/clients/generic_client.rs

```diff
@@ -134,31 +166,51 @@ where
 
 #[cfg(test)]
 mod tests {
-    use super::api::MockDexagApi;
     use super::*;
+    use crate::models::TokenId;
     use crate::price_estimation::data::TokenBaseInfo;
-    use crate::util::FutureWaitExt as _;
+    use anyhow::anyhow;
     use lazy_static::lazy_static;
     use mockall::{predicate::*, Sequence};
+    use std::sync::Once;
+
+    // Expectations for static methods are shared between multiple tests.
+    // https://docs.rs/mockall/0.6.0/mockall/#static-methods
+    // Here we create a shared context once and keep it in memory. Creating
+    // new contexts for each test would cause race conditions on execution.
+    fn initialize_mockapi_context() {
+        static SETUP_ONCE: Once = Once::new();
+
+        SETUP_ONCE.call_once(|| {
+            let ctx = MockApi::stable_coin_symbol_context();
+            ctx.expect().returning(|| "DAI".to_string());
+            std::mem::forget(ctx);
+        });
+    }
```

Comment:
😭 So much complication to mock a non instance method? That's a shame. I briefly looked into what would happen if we made STABLECOIN_SYMBOL an associated constant but that doesn't seem supported by mockall.

Passing the stablecoun symbol into the constructor of the GenericClient would work but separate the logic of what the stablecoin is away from the API. 

So leaving this is fine, i guess.